Libraries/Lists/__tests__/VirtualizedList-test.js

```diff
@@ -273,4 +273,92 @@ describe('VirtualizedList', () => {
       }),
     );
   });
+
+  it('should be called onEndReached correctly', () => {
+    function appendNewItems(items, count) {
+      const nextId = (items.length > 0 ? items[items.length - 1].id : 0) + 1;
+
+      for (let loop = 1; loop <= count; loop++) {
+        const id = nextId + loop;
+        items.push({
+          id: id,
+          key: `k${id}`,
+        });
+      }
+
+      return items;
+    }
+
+    let scrollTimeStamp = 0;
+    function scrollList(instance, items, y) {
+      scrollTimeStamp += 1000;
+
+      const nativeEvent = {
+        contentOffset: {y, x: 0},
+        layoutMeasurement: {width: 300, height: 600},
+        contentSize: {width: 300, height: items.length * ITEM_HEIGHT},
+        zoomScale: 1,
+        contentInset: {right: 0, top: 0, left: 0, bottom: 0},
+      };
+
+      instance._onScroll({
+        timeStamp: scrollTimeStamp,
+        nativeEvent,
+      });
+    }
+
+    const ITEM_HEIGHT = 100;
+    const APPENDED_ITEM_COUNT = 10;
+    const data = appendNewItems([], 10);
+    const onEndReached = jest.fn(function() {
+      appendNewItems(data, APPENDED_ITEM_COUNT);
+    });
+
+    const props = {
+      data,
+      renderItem: ({item}) => <item value={item.key} />,
+      getItem: (items, index) => items[index],
+      getItemCount: items => items.length,
+      getItemLayout: (items, index) => ({
+        length: ITEM_HEIGHT,
+        offset: ITEM_HEIGHT * index,
+        index,
+      }),
+      onEndReached,
+    };
+
+    const component = ReactTestRenderer.create(<VirtualizedList {...props} />);
+    const instance = component.getInstance();
+
+    // It should not be called after the initial rendering
+    expect(onEndReached).not.toHaveBeenCalled();
+    expect(data.length).toBe(10);
+
+    // It should be called once after scrolling by 200
+    scrollList(instance, data, 200);
+
+    expect(onEndReached).toHaveBeenCalledTimes(1);
+    expect(onEndReached).toHaveBeenLastCalledWith({
+      distanceFromEnd: 200,
+    });
+    expect(data.length).toBe(20);
+
+    // It should not be called after scrolling to the top
+    scrollList(instance, data, 0);
+
+    expect(onEndReached).toHaveBeenCalledTimes(1);
+    expect(data.length).toBe(20);
+
+    // It should not be called after scrolling back on the screen
```

Comment:
I don't understand what this is testing. What does "scrolling back on the screen" mean?

There are a lot of expectations in this test, would it be possible to split it into multiple smaller test cases that would each cover individual cases?

Reply:
@vonovak It was a test for this case(scroll down -> scroll to the top -> scroll down again). This test doesn't seem so meaningful. :)

I separated the test because I agree with you. Thanks for your review.